src/prefect/tasks/google/bigquery.py

```diff
@@ -271,3 +272,98 @@ def run(
+        - schema (List[bigquery.SchemaField], optional): the schema to use when creating the table
+        - clustering_fields (List[str], optional): a list of fields to cluster the table by
+        - time_partitioning (bigquery.TimePartitioning, optional): a `bigquery.TimePartitioning` object specifying
+            a partitioninig of the newly created table
+        - **kwargs (optional): additional kwargs to pass to the `Task` constructor
+    """
+
+    def __init__(
+        self,
+        project: str = None,
+        credentials_secret: str = None,
+        dataset: str = None,
+        table: str = None,
+        schema: List[bigquery.SchemaField] = None,
+        clustering_fields: List[str] = None,
+        time_partitioning: bigquery.TimePartitioning = None,
+        **kwargs
+    ):
+        self.project = project
+        self.credentials_secret = credentials_secret or "GOOGLE_APPLICATION_CREDENTIALS"
+        self.dataset = dataset
+        self.table = table
+        self.schema = schema
+        self.clustering_fields = clustering_fields
+        self.time_partitioning = time_partitioning
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("project", "credentials_secret", "dataset", "table", "schema")
+    def run(
+        self,
+        project: str = None,
+        credentials_secret: str = None,
+        dataset: str = None,
+        table: str = None,
+        schema: List[bigquery.SchemaField] = None,
+    ):
+        """
+        Run method for this Task.  Invoked by _calling_ this Task within a Flow context, after initialization.
+
+        Args:
+            - project (str, optional): the project to initialize the BigQuery Client with; if not provided,
+                will default to the one inferred from your credentials
+            - credentials_secret (str, optional): the name of the Prefect Secret containing a JSON representation
+                of your Google Application credentials; defaults to `"GOOGLE_APPLICATION_CREDENTIALS"`
+            - dataset (str, optional): the name of a dataset in which the table will be created
+            - table (str, optional): the name of a table to create
+            - schema (List[bigquery.SchemaField], optional): the schema to use when creating the table
+
+        Returns:
+            - None
+
+        Raises:
+            - SKIP: a `SKIP` signal if the table already exists
+        """
+        creds = Secret(credentials_secret).get()
+        credentials = Credentials.from_service_account_info(creds)
+        project = project or credentials.project_id
+        client = bigquery.Client(project=project, credentials=credentials)
+
+        try:
+            dataset_ref = client.get_dataset(dataset)
+        except NotFound:
+            dataset_ref = client.create_dataset(dataset)
+
+        table_ref = dataset_ref.table(table)
+        try:
+            client.get_table(table_ref)
+            raise SKIP(
```

Comment:
I want to make an argument that this should be a success, not a skip, since it did the thing the task claimed: ensured the table exists. Otherwise, a downstream task that requires the table to exist would have to set `skip_on_upstream_skip=False`, which feels like an unnatural init arg.

On the other hand, it is in fact skipping the bulk of its logic. 

Not sure I have a valuable opinion here...

Reply:
That's true, I agree with your reasoning.  My goal with raising a signal was so this task could be seamlessly included into a workflow which runs regularly for the sake of preserving _all_ logic.  Given that a `SUCCESS` signal is much more appropriate.  

I think my initial reasoning was so the task would have a noticeably different state if the table already exists, but I spaced on the `skip_on_upstream_skip` behavior.

Reply:
Updated